File: API/Entities/Member.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace API.Entities;

public class Member
{
   public string Id {get; set; } = null!;
   public required string DisplayName {get; set;}
   public string? ImageUrl {get; set;}
   public DateOnly DateOfBirth {get;set;}
   public required string Country {get; set;}
   public required string City {get;set;}
   public required string Gender {get; set;}
   public string? Description {get;set;} 
   public DateTime DateCreated {get; set;} = DateTime.UtcNow;
   public DateTime LastActive {get;set;} = DateTime.UtcNow;
   public List<Photo> Photos {get;set;} = [];
   
   [ForeignKey(nameof(Id))]
   public AppUser User {get;set;} = null!;
  [JsonIgnore]
   public List<MemberLike> LikedMember {get;set;} = [];
[JsonIgnore]
   public List<MemberLike> LikedByMember {get;set;} = [];
   public List<Message> MessagesSent {get;set;} = [];

   public List<Message> MessagesRecieved {get;set;}=[];

}

File: client/src/types/Member.ts

export type Members = {
  id: string
  displayName: string
  imageUrl?: string
  dateOfBirth: string
  country: string
  city: string
  gender: string
  description?: string
  dateCreated: string
  lastActive: string
}


export type Photo = {
  id: number
  url: string
  publicId?: string
  memberId: string
}

export type UpdateMember = {
  displayName : string
  description : string
  city : string
  country : string
}

export class MemberParam {
  gender? : string
  minAge = 18
  maxAge = 100
  pageNumber = 1
  pageSize = 10
  orderBy = "lastActive"
}

export class MemberLikeParam{
   predicate = 'liked'
   pageNumber = 1
   pageSize = 10 
}